Makefile

```diff
@@ -29,27 +29,19 @@ help: ## Show this help message
 venv: ## Create Python virtual environment
 	@if [ ! -d "$(VENV_DIR)" ]; then \
 		printf "$(BLUE)Creating virtual environment...$(RESET)\n"; \
-		$(UV) venv $(VENV_DIR); \
+		$(PYTHON) -m venv $(VENV_DIR); \
 		printf "$(GREEN)✅ Virtual environment created$(RESET)\n"; \
 	else \
 		printf "$(YELLOW)Virtual environment already exists$(RESET)\n"; \
 	fi
 
-# Install uv
-.PHONY: uv
-uv: venv ## Install uv
-	@if [ ! -f "$(VENV_DIR)/bin/uv" ]; then \
-		printf "$(BLUE)Installing uv...$(RESET)\n"; \
-		$(VENV_PYTHON) -m ensurepip --upgrade; \
-		$(VENV_PYTHON) -m pip install uv; \
-		printf "$(GREEN)✅ uv installed$(RESET)\n"; \
-	fi
-
 # Install requirements
 .PHONY: pip-install-test
-pip-install-test: uv ## Install package dependencies
+pip-install-test: venv ## Install package dependencies
 	@printf "$(BLUE)Installing dependencies...$(RESET)\n"
-	$(UV) pip install -r requirements-test.txt
+	$(VENV_PYTHON) -m pip install --upgrade pip
+	$(VENV_PYTHON) -m pip install -r requirements-test.txt
+	$(VENV_PYTHON) -m pip install -e .
```

Comment:
uv can be used once the venv is installed. it is preferred for performance.  keep use of python -m venv ONLY for the creation of the venv.  after that, use uv instead of pip